pkg/highlight/parser.go

```diff
@@ -476,10 +476,17 @@ func parseRegion(group string, regionInfo map[interface{}]interface{}, prevRegio
 		r.limitGroup = r.group
 	}
 
-	r.rules, err = parseRules(regionInfo["rules"].([]interface{}), r)
+	// rules are optional
+	if rules, ok := regionInfo["rules"]; ok {
+		r.rules, err = parseRules(rules.([]interface{}), r)
+		if err != nil {
+			return nil, err
+		}
+	}
 
-	if err != nil {
-		return nil, err
+	if r.rules == nil {
+		// allow empty rules
+		r.rules = new(rules)
```

Comment:
nit: It seems a bit more idiomatic to use `&rules{}` instead of `new(rules)` in such cases. (Yeah it was me who first used `new(rules)` in #3213. We can change it to `&rules{}` there as well.)